### ISSUE TITLE
Unable to install markupsafe in amazon ec2 redhat virtual env 

### ISSUE THREAD

**Issue by sureshimp766** [pallets/markupsafe#85]
I am unable to install the markupsafe in the virtual env on amazon ec2 
Linux 4.9.51-10.52.amzn1.x86_64 #1 SMP Fri Sep 29 01:16:19 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

I am using python 2.7.12 and virtualenv 15.1.0 and pip 9.0.1 . I first install Flask using "pip install Flask"
which says 
Successfully installed Flask-0.12.2 Jinja2-2.10 MarkupSafe-1.0 Werkzeug-0.12.2 click-6.7 itsdangerous-0.24
pip freeze shows 

```
click==6.7
Flask==0.12.2
itsdangerous==0.24
Jinja2==2.10
Werkzeug==0.12.2
```

But it does not show the MarkupSafe. I tried install separately, but same fate. 
please help.

**Comment by ThiefMaster** [comment]
You need to post the full output when trying to install it, not just a snippet. Ideally, use `pip -v install markupsafe` to enable verbose output, then post that command's output.

**Comment by sureshimp766** [comment]
Thank you for kind response. Here you go,
(trysmautoenv) [pcdeuser@ip-172-31-81-178 ~]$ pip -v install markupsafe
Collecting markupsafe
  1 location(s) to search for versions of markupsafe:
  * https://pypi.python.org/simple/markupsafe/
  Getting page https://pypi.python.org/simple/markupsafe/
  Looking up "https://pypi.python.org/simple/markupsafe/" in the cache
  Current age based on date: 82
  Freshness lifetime from max-age: 600
  Freshness lifetime from request max-age: 600
  The response is "fresh", returning cached response
  600 > 82
  Analyzing links from page https://pypi.python.org/simple/markupsafe/
    Found link https://pypi.python.org/packages/00/00/d268f88c561a92e35e1df96e68dbec2210926366c79fd426b1d9ee99b020/MarkupSafe-0.12.tar.gz#md5=edd4b6950f6d4ddc6e23fa322f8f0f53 (from https://pypi.python.org/simple/markupsafe/), version: 0.12
    Found link https://pypi.python.org/packages/00/52/33fd9769edf01ffab9e3fd321333fe4d9f7ac9cb0b17ce9524abcb64e089/MarkupSafe-0.15.tar.gz#md5=4e7c4d965fe5e033fa2d7bb7746bb186 (from https://pypi.python.org/simple/markupsafe/), version: 0.15
    Found link https://pypi.python.org/packages/04/d0/21c43bb0a9c9b31c8bfeb3676e12ec0aae2b71632497b6bd6505c980a38a/MarkupSafe-0.17.tar.gz#md5=61d5edc130e3242cfaff9fd9fdefde8f (from https://pypi.python.org/simple/markupsafe/), version: 0.17
    Found link https://pypi.python.org/packages/36/83/400b5f58bfa88f96bc68ccd963990c80549a2825d0777bd838f03d8f0b42/MarkupSafe-0.9.2.tar.gz#md5=69b72d1afdd9e808f9c1ef65f819c7a6 (from https://pypi.python.org/simple/markupsafe/), version: 0.9.2
    Found link https://pypi.python.org/packages/3b/45/dfad433343922655477760d506b058f96a783e8c0ada7b8d0602ff2beace/MarkupSafe-0.9.1.tar.gz#md5=7f469f664b565b440f099997255e6d3c (from https://pypi.python.org/simple/markupsafe/), version: 0.9.1
    Found link https://pypi.python.org/packages/44/ee/37bd0844418acd5fb9c588aae835f38b9b7c4cd0c9b95c78c8ebb74bb65a/MarkupSafe-0.16.tar.gz#md5=fe88012f7fe75252efd26cd25a100851 (from https://pypi.python.org/simple/markupsafe/), version: 0.16
    Found link https://pypi.python.org/packages/4d/de/32d741db316d8fdb7680822dd37001ef7a448255de9699ab4bfcbdf4172b/MarkupSafe-1.0.tar.gz#md5=2fcedc9284d50e577b5192e8e3578355 (from https://pypi.python.org/simple/markupsafe/), version: 1.0
    Found link https://pypi.python.org/packages/58/18/646cbd99909a0c86eb7b9c0277b88cb54f3e2619f6115f48199a4accde97/MarkupSafe-0.22.tar.gz#md5=cb3ec29fd5361add24cfd0c6e2953b3e (from https://pypi.python.org/simple/markupsafe/), version: 0.22
    Found link https://pypi.python.org/packages/61/9d/195c0e2e9efcba7b5f2cad9d01ea7311e20358a7d1c8a928fb481eecad09/MarkupSafe-0.13.tar.gz#md5=f44b8f4bba98b3e66884158f97585829 (from https://pypi.python.org/simple/markupsafe/), version: 0.13
    Found link https://pypi.python.org/packages/6f/c3/7adab2342dc7f14d32943505c9bd163d93a16fa117b8fee82e48f3529fb5/MarkupSafe-0.20.tar.gz#md5=7da066d9cb191a70aa85d0a3d43565d1 (from https://pypi.python.org/simple/markupsafe/), version: 0.20
    Found link https://pypi.python.org/packages/8e/90/da092a12fb96e0c4cacc279d1f92819ae82bfa291e0a03afe8059518e91a/MarkupSafe-0.19.tar.gz#md5=ccb3f746c807c5500850987006854a6d (from https://pypi.python.org/simple/markupsafe/), version: 0.19
    Found link https://pypi.python.org/packages/98/cf/197c3b0f73224b84eb419a967f87565bcc0b0c1147012397e6bd2d45e253/MarkupSafe-0.18.tar.gz#md5=f8d252fd05371e51dec2fe9a36890687 (from https://pypi.python.org/simple/markupsafe/), version: 0.18
    Found link https://pypi.python.org/packages/be/30/7dfaf5f61e9648a9bb97ecd0099da2546ea78fe93cb9c777030c6d056c0d/MarkupSafe-0.9.tar.gz#md5=10a7c6bf81f91f280ba9e76e32e01c94 (from https://pypi.python.org/simple/markupsafe/), version: 0.9
    Found link https://pypi.python.org/packages/c0/41/bae1254e0396c0cc8cf1751cb7d9afc90a602353695af5952530482c963f/MarkupSafe-0.23.tar.gz#md5=f5ab3deee4c37cd6a922fb81e730da6e (from https://pypi.python.org/simple/markupsafe/), version: 0.23
    Found link https://pypi.python.org/packages/c6/33/963f57460372f8401e1f9aa32c8ed8f07a5cae0f11024f5ff1d4cb3576c8/MarkupSafe-0.21.tar.gz#md5=fde838d9337fa51744283f46a1db2e74 (from https://pypi.python.org/simple/markupsafe/), version: 0.21
    Found link https://pypi.python.org/packages/cf/72/6ad21b64f007dba3d211a973f37e3bf0bffae88c4efb868999fd2c94497a/MarkupSafe-0.14.tar.gz#md5=68c6ea6df002649218d24cf490a8ca82 (from https://pypi.python.org/simple/markupsafe/), version: 0.14
    Found link https://pypi.python.org/packages/ee/51/df4a007a19f3de1b733adac77da11fa45f1ec5089549a7cc2d67195bb2c1/MarkupSafe-0.11.tar.gz#md5=48d445941c16d6aa55caf8e148fc0911 (from https://pypi.python.org/simple/markupsafe/), version: 0.11
    Found link https://pypi.python.org/packages/f1/a7/39b74c3c7b46531842d2a9a55b7d6bf5b44851786d131d0fa76dbb4abf44/MarkupSafe-0.9.3.tar.gz#md5=99dccbeb83302f6b593d382098b3b43a (from https://pypi.python.org/simple/markupsafe/), version: 0.9.3
  Using version 1.0 (newest of versions: 0.9, 0.9.1, 0.9.2, 0.9.3, 0.11, 0.12, 0.13, 0.14, 0.15, 0.16, 0.17, 0.18, 0.19, 0.20, 0.21, 0.22, 0.23, 1.0)
  Using cached wheel link: file:///home/pcdeuser/.cache/pip/wheels/88/a7/30/e39a54a87bcbe25308fa3ca64e8ddc75d9b3e5afa21ee32d57/MarkupSafe-1.0-cp27-cp27mu-linux_x86_64.whl
Installing collected packages: markupsafe

Successfully installed markupsafe-1.0
Cleaning up...

**Comment by davidism** [comment]
> Successfully installed markupsafe-1.0

Looks like it's installed. Does `pip list` include it now?

**Comment by oDoraemon** [comment]
It comes to me with the amz ec2 rehat too.
run pip install, the output said it's installed. But pip list did not list it.
and run pip uninstall, prompt MarkupSafe not installed.

![_20180403170121](https://user-images.githubusercontent.com/7870718/38239952-ca4c925c-3760-11e8-8598-c1098f8da400.png)

But I can find the MarkupSafe dir and egg-info files under .venv/lib64/dist-packages.

**Comment by oDoraemon** [comment]
OK, Here is my case.
pip install fail somehow , i give it up.
Then I download the tar ball, use 'python setup install'. 
Got prompt to set PYTHONPATH. The package can only install under lib64/python2.7/dist-packages, so I export PYTHONPATH=blabla/lib64/.../dist-packages
After that, I can see MarkupSafe in pip list and able to import in python shell.
But my flask app (base on wsgi) still report unable import markupsafe error.
So I wander it may be the env PATH problem. vi my app.wsgi file, and addsitedir PYTHONPATH(eg. lib64/../dist-packages). httpd -k restart , BAM!  Website work again.
Oh God! It take me a whole day !
I have no idea why it happen. I swear I never change any environ things. I just change some of my app files, but fail to lauch my app. I recover from backup, and then this problem came to me. Wired

**Comment by davidism** [comment]
Closing as I can't reproduce this but it seems like you worked around it. This should also be solved by #45.

**Comment by davidism** [comment]
PyPI has wheels for MarkupSafe 1.1.0.
